src/services/mentors.js

```diff
@@ -551,27 +552,38 @@ module.exports = class MentorsHelper {
 	 * @param {Boolean} isAMentor 				- user mentor or not.
 	 * @returns {JSON} 							- profile details
 	 */
-	static async read(id, orgId, userId = '', isAMentor = '') {
+	static async read(id, orgId, userId = '', isAMentor = '', roles = '') {
 		try {
-			if (userId !== '' && isAMentor !== '') {
+			let requestedMentorExtension = false
+			if (userId !== '' && isAMentor !== '' && roles !== '') {
 				// Get mentor visibility and org id
-				let requstedMentorExtension = await mentorQueries.getMentorExtension(id, [
-					'mentor_visibility',
-					'mentee_visibility',
-					'organization_id',
-					'visible_to_organizations',
-				])
+				requestedMentorExtension = await mentorQueries.getMentorExtension(id)
+
+				const validateDefaultRules = await validateDefaultRulesFilter({
```

Comment:
@nevil-mathew are you applying the rules after getting the data ?

Reply:
Yes, because we need the data to validate.